api/src/opentrons/protocol_engine/state/labware_origin_math/stackup_origin_to_labware_origin.py

```diff
@@ -157,24 +170,48 @@ def _get_stackup_origin_to_lw_origin(
             stackup_lw_info_top_to_bottom=remaining_lw_defs_locs_top_to_bottom,
             underlying_ancestor_definition=underlying_ancestor_definition,
             module_parent_to_child_offset=module_parent_to_child_offset,
-            slot_name=slot_name,
             deck_definition=deck_definition,
+            slot_name=slot_name,
             is_topmost_labware=False,
         )
     else:
-        raise errors.LabwareNotOnDeckError(
+        raise LabwareNotOnDeckError(
             "Cannot access labware since it is not on the deck. "
             "Either it has been loaded off-deck or its been moved off-deck."
         )
 
 
+def _get_underlying_ancestor_orientation(
+    underlying_ancestor_definition: LabwareStackupAncestorDefinition,
+    slot_id: DeckSlotName,
+) -> ModuleOrientation:
+    if isinstance(underlying_ancestor_definition, ModuleDefinition):
+        orientation = underlying_ancestor_definition.orientation.get(str(slot_id))
```

Comment:
Nit: Can we do `.id` instead of `str()`, for explicitness?

```suggestion
        orientation = underlying_ancestor_definition.orientation.get(slot_id.id)
```